onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -257,6 +257,11 @@ class PlannerImpl {
 
   // Find if there exists some input tensor that we can use in-place for output_arg_num-th input in the node.
   bool FindReusableInput(const onnxruntime::Node& node, int output_arg_num, OrtValueIndex* reusable_input) {
+    auto p_next_node = node.OutputNodesBegin();
+    if (p_next_node != node.OutputNodesEnd() && p_next_node->OpType() == "YieldOp") {
+      return false;
+    }
```

Comment:
@SherlockNoMad This will essentially ensure we are always allocating memory for YieldOp's inputs which is fine because now we won't have the re-use issue downstream and the ownership can be safely transfered to PyTorch but have you measured how this impacts max batch size on our models?

Reply:
This change would affect the memory for the models that ends with a op that uses .alias or .may-in-place (Relu, Reshape, Squeeze etc... )

In the UT, we composed a model like this: Reshape feeding into YieldOp
![image](https://user-images.githubusercontent.com/9906745/119048369-feac1480-b973-11eb-9df2-70b22d9d8431.png)

In such models, it's inevitable that we need to reserve a buffer for YieldOp's input.

Reply:
Still not fully understanding the ownership/re-use issue. 

If you wish to transfer ownership to an external system, I would have naively expected that they YieldOp does an internal allocation an CopyTensor call to make the ownership semantics clearer. I assume you don't do that to save a copy.

As for the re-use, is there also an edge for the value that is the YieldOp input to another node downstream of the YieldOp? e.g. NodeA produces output B, which is an input to both the YieldOp and NodeC. So NodeA creates B, you hit the YieldOp, PyTorch changes B, and after execution of the YieldOp completes we execute NodeC which tries to use a now invalid B? 

If so, you only need to avoid re-use if there are two edges for the value don't you? 

Not quite understanding how the corruption would happen if that wasn't the case, given that B would only be used by the YieldOp, so if PyTorch changed B nothing downstream would be affected by that change.

Reply:
Yes, the scenario you described here is exactly the the corner case I mentioned in the comment. 

Indeed, we are trying to avoid the memcpy, coz the forward output, in many times, is the prediction tensor, which could be very large. 

We are considering to introduce a copy for the fetches of PartialGraphExecutor. 
By the time the tensor is returned to pytorch, if there is still downstream dependency, we will return a copy of the tensor.